lib/classes/Variables.test.js

```diff
@@ -76,6 +74,33 @@ describe('Variables', () => {
         expect(serverless.service.resources.bar).to.equal(barValue);
       });
     });
+
+    it('should delete null environment variables', () => {
+      const serverless = new Serverless();
+
+      serverless.service.provider.environment = {
+        SOME_VAR: 'some-value',
+        OTHER_VAR: null,
+      };
+      serverless.service.functions = {
+        hello: {
+          handler: 'hello.js',
+          environment: {
+            VAR_ONE: 'one',
+            VAR_TWO: null,
+          },
+        },
+      };
+
+      return serverless.variables.populateService().then(() => {
+        /* eslint-disable no-unused-expressions */
```

Comment:
Interesting.  Yeah, to your point, I'd guess there are likely to be more false positives for `no-unused-expressions` than actual positives in test files.   And, just as a general principal, it's better to configure your linter to work with how you want your code to be, as opposed to changing your code to fit the default constraints imposed by your linter.  

All that said, I might be underestimating the value of `no-unused-expressions` in test files, and `.all()`-ing the `expects` doesn't seem like a terrible workaround.